src/server/page_server.cpp

```diff
@@ -245,36 +252,30 @@ page_server::connection_handler::on_log_page_read_result (const LOG_PAGE *log_pa
 }
 
 void
-page_server::connection_handler::on_data_page_read_result (const FILEIO_PAGE *io_page, int error_code)
+page_server::connection_handler::on_data_page_read_result (tran_server_conn_t::sequenced_payload &&sp,
+    const FILEIO_PAGE *io_page, int error_code)
 {
   std::string message;
   if (error_code != NO_ERROR)
     {
-      char buffer[sizeof (int)];
-      std::memcpy (buffer, &error_code, sizeof (error_code));
-      message = std::string (buffer, sizeof (int));
-    }
-  else
-    {
-      char buffer[IO_MAX_PAGE_SIZE];
-      std::memcpy (buffer, io_page, db_io_page_size ());
-      message = std::string (buffer, db_io_page_size ());
-    }
-
-  if (prm_get_bool_value (PRM_ID_ER_LOG_READ_DATA_PAGE))
-    {
-      if (error_code == NO_ERROR)
+      message.append (reinterpret_cast<const char *> (&error_code), sizeof (error_code));
```

Comment:
We can unify the approaches for data and log pages. Either always bundle the error code, or add alternatively error code or the actual page. Currently, in both cases, any errors are fatal. But, can it be that some retry/failover mechanism is implemented when more than one page servers would be present?

Reply:
> We can unify the approaches for data and log pages. Either always bundle the error code, or add alternatively error code or the actual page.

Yes, you're right. I'll make the change to always bundle them together.